macos/Sources/App/macOS/AppDelegate.swift

```diff
@@ -245,7 +245,14 @@ class AppDelegate: NSObject,
         // This probably isn't fully safe. The isEmpty check above is aspirational, it doesn't
         // quite work with SwiftUI because windows are retained on close. So instead we check
         // if there are any that are visible. I'm guessing this breaks under certain scenarios.
-        if (windows.allSatisfy { !$0.isVisible }) { return .terminateNow }
+        if (windows.allSatisfy { !$0.isVisible }) {
```

Comment:
Huh, this check is actually very problematic in general. If I'm reading this right, having hidden windows (Cmd+H) would bypass close confirmation as well.

I suggest we look into this line and redo this to maybe be based on `BaseTerminalController` inspection.

Reply:
I was browsing the history of this line and its very suspicious, I think its to workaround an only retain cycle we had. I don't think this visible check needs to apply anymore. BUT, I don't want to rock the boat for 1.1.1 so I'm going to add a note to look at for 1.2 and get this in.